scripts/build_package.py

```diff
@@ -554,70 +342,52 @@ def build_package(repository_root, build_package_config, env=None):
     copy_tree(source, target)
 
     # Thift js.
-    thrift_dep = vendor_projects['thrift']
-    thrift_root = os.path.join(repository_root, thrift_dep.get('directory'))
+    thrift_root = os.path.join(npm_modules, 'thrift')
     thift_js_files = os.path.join(thrift_root, 'lib', 'js', 'src')
     target = os.path.join(package_root, package_layout['js_thrift'])
-    copy_tree(thift_js_files, target)
+    copy_tree(thift_js_files, target, None, yui_compress_jar)
 
     # CodeMirror.
-    codemirror_dep = vendor_projects['codemirror']
-    codemirror_root = os.path.join(repository_root,
-                                   codemirror_dep.get('directory'))
+    codemirror_root = os.path.join(npm_modules, 'codemirror')
     target = os.path.join(package_root,
                           package_layout['web_client_codemirror'])
     copy_tree(codemirror_root, target)
 
-    # HighlightJs.
-    highlightjs_dep = vendor_projects['highlightjs']
-    highlightjs_root = os.path.join(repository_root,
-                                    highlightjs_dep.get('directory'))
-    target = os.path.join(package_root,
-                          package_layout['web_client_highlightjs'])
-    copy_tree(highlightjs_root, target)
-
-    # HighlightJs_css.
-    highlightjs_css_dep = vendor_projects['highlightjs_css']
-    highlightjs_css_root = os.path.join(repository_root,
-                                        highlightjs_css_dep.get('directory'))
-    target = os.path.join(package_root,
-                          package_layout['web_client_highlightjs'])
-    target = os.path.join(target, 'css')
-    copy_tree(highlightjs_css_root, target)
-
-    # Dojo.
-    dojo_dep = vendor_projects['dojotoolkit']
-    file_url = dojo_dep['source_package']['url']
-    url_data = urlparse.urlparse(file_url)
-    _, file_name = os.path.split(url_data.path)
-    head, _ = file_name.split('.tar.gz')
-
-    dojo_root = os.path.join(repository_root, dojo_dep.get('directory'))
-    dojo_root = os.path.join(dojo_root, head)
-    target = os.path.join(package_root, package_layout['web_client_dojo'])
-    copy_tree(dojo_root, target)
+    if minify:
```

Comment:
A comment would be nice, that each js file which should be minified should be listed here.